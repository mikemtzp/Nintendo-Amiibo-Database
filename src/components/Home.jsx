const Home = () => (
  <section>
    <div>
      <img src="#" alt="test" />
      Welcome! Select a category to filter and see all the information available about any Amiibo!
      <div className="header-division" />
    </div>
    <ul />
  </section>
);

export default Home;
