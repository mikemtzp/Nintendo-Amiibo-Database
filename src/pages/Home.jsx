import React from 'react';
import { useSelector } from 'react-redux';
import Category from '../components/Amiibo';
import logo from '../assets/images/Amiibo_main.jpg';

const Home = () => {
  const { amiibos } = useSelector((state) => state.amiibos);
  const amiiboList = Object.keys(amiibos);

  return (
    <section>
      <div className="header">
        <img src={logo} alt="All amiibos" className="header-image" />
        <p className="header-text">
          Welcome! Select a category to filter with the searchbar
          above and see all the information available about any Amiibo!
        </p>
      </div>
      <ul>
        {
          amiiboList ? amiiboList.map((amiibo) => (
            <Category
              series={amiibos[amiibo].series}
              character={amiibos[amiibo].character}
              game={amiibos[amiibo].game}
              image={amiibos[amiibo].image}
              name={amiibos[amiibo].name}
              type={amiibos[amiibo].type}
              key={amiibo}
            />
          )) : <li>No categories available</li>
        }
      </ul>
    </section>
  );
};

export default Home;
