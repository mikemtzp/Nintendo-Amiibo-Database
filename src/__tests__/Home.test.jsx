import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import Home from '../pages/Home';
import store from '../redux/configStore';

describe('Renders Home tests', () => {
  it('Renders Home component', () => {
    const tree = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Renders home seccions', () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(screen.getByText(/All Game Series/i)).toBeInTheDocument();
    expect(document.querySelector('.header-text')).toBeInTheDocument();
    expect(document.querySelector('.header-amiibocontainer')).toBeInTheDocument();
  });
});
