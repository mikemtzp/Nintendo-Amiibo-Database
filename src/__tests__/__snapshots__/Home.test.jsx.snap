// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home tests Renders Home component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="header"
        >
          <img
            alt="All amiibos"
            class="header-image"
            src="Amiibo_main.jpg"
          />
          <p
            class="header-text"
          >
            Welcome! Select a category to filter with the search bar below and see all the information available about any amiibo!
          </p>
          <select
            class="header-filter"
          >
            <option
              value="all"
            >
              All Game Series
            </option>
          </select>
        </div>
        <ul
          class="header-amiibocontainer"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="header"
      >
        <img
          alt="All amiibos"
          class="header-image"
          src="Amiibo_main.jpg"
        />
        <p
          class="header-text"
        >
          Welcome! Select a category to filter with the search bar below and see all the information available about any amiibo!
        </p>
        <select
          class="header-filter"
        >
          <option
            value="all"
          >
            All Game Series
          </option>
        </select>
      </div>
      <ul
        class="header-amiibocontainer"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
